dds/DCPS/StaticIncludes.h

```diff
@@ -41,8 +41,7 @@
 #    include "transport/rtps_udp/RtpsUdp.h"
 #  else
 // tcp is needed for InfoRepoDiscovery
-#    include "transport/tcp/Tcp.h"
-#    include "InfoRepoDiscovery/InfoRepoDiscovery.h"
+#    include "RTPS/RtpsDiscovery.h"
```

Comment:
Remove comment 1 line up -- it no longer applies

Refactor common lines of code on each side of the `#if` / `#else`